deepchecks/core/check_json.py

```diff
@@ -0,0 +1,114 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing the check results classes."""
+# pylint: disable=super-init-not-called
+import io
+from typing import Dict, List, Union
+
+import jsonpickle
+import pandas as pd
+import plotly
+
+from deepchecks.core.check_result import CheckFailure, CheckResult
+from deepchecks.core.condition import (Condition, ConditionCategory,
+                                       ConditionResult)
+
+__all__ = [
+    'CheckResultJson',
+    'CheckFailureJson',
+]
+
+
+class FakeCheck:
+    def __init__(self, metadata: Dict):
+        self._metadata = metadata
+
+    def metadata(self, *args, **kwargs):  # pylint: disable=unused-argument
+        return self._metadata
+
+    def name(self):
+        return self._metadata['name']
+
+
+class CheckResultJson(CheckResult):
```

Comment:
Why do we need `CheckResultJson` if we have the `FakeCheck`?
can we just initialize `CheckResult` with `FakeCheck` instance?

Reply:
we still have some different functions and I prefer having an init instead of seperate util functions

Reply:
why? that we discussed about, having general `deepchecks.from_json`

Reply:
and I don't see any different functions except the pass in process_conditions

Reply:
@matanper it is still a different function, and there is a global from json